src/UmbracoFileSystemProviders.Azure/FileSystemVirtualFile.cs

```diff
@@ -68,7 +69,7 @@ public override Stream Open()
                 // Add Accept-Ranges header to fix videos not playing on Safari
                 HttpContext.Current.Response.AppendHeader("Accept-Ranges", "bytes");
 
-                IFileSystem azureBlobFileSystem = FileSystemProviderManager.Current.GetUnderlyingFileSystemProvider("media");
+                IFileSystem azureBlobFileSystem = Current.MediaFileSystem.Unwrap();
```

Comment:
This looks absolutely mental to me. `Unwrap()`? The `IFileSystem` should be injectable.